crates/sui-move-build/src/linters/freeze_wrapped.rs

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This analysis flags freezing instances of structs containing, transitively or not, other structs
+//! with the key ability.
+
+use move_compiler::{
+    diag,
+    diagnostics::codes::{custom, DiagnosticInfo, Severity},
+    expansion::ast as E,
+    hlir::{ast as H, visitor::HlirVisitor},
+    parser::ast as P,
+    shared::{unique_map::UniqueMap, CompilationEnv, Identifier},
+};
+use move_ir_types::location::*;
+
+use super::{LinterDiagCategory, LINTER_DEFAULT_DIAG_CODE, LINT_WARNING_PREFIX};
+
+const FREEZE_KEY_DIAG: DiagnosticInfo = custom(
+    LINT_WARNING_PREFIX,
+    Severity::Warning,
+    LinterDiagCategory::FreezeWrapped as u8,
+    LINTER_DEFAULT_DIAG_CODE,
+    "attempting to freeze wrapped objects",
+);
+
+const FREEZE_FUNCTIONS: &[(&str, &str, &str)] = &[
+    ("sui", "transfer", "public_freeze_object"),
+    ("sui", "transfer", "freeze_object"),
+];
```

Comment:
Some/all of these names are defined as constants now I believe, so probably best to use those here.

Reply:
Fixed! An artifact of developing this in parallel with another PR that introduced these constants...